[imgui:x64-windows] build failure

**Host Environment**
- OS: Windows 10 Enterprise 2004
- Compiler: MSVC 14.27.29110
- Vcpkg version: 2020.06.15-nohash
- Vulkan SDK: 1.2.148.1

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install imgui[*]`

**Failure logs** 
````
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:139 (message):
    Command failed: D:/vcpkg/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory: D:/vcpkg/buildtrees/imgui/x64-windows-dbg
    See logs for more information:
      D:\vcpkg\buildtrees\imgui\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/imgui/portfile.cmake:44 (vcpkg_install_cmake)
  scripts/ports.cmake:79 (include)


Error: Building package imgui:x64-windows failed with: BUILD_FAILED
````

D:\vcpkg\buildtrees\imgui\install-x64-windows-dbg-out.log:
```
[1/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_opengl2.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_opengl2.cpp
[2/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_vulkan.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_vulkan.cpp
[3/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_opengl3.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_opengl3.cpp
[4/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\misc\freetype\imgui_freetype.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\misc\freetype\imgui_freetype.cpp
[5/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\imgui_demo.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\imgui_demo.cpp
[6/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\misc\cpp\imgui_stdlib.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\misc\cpp\imgui_stdlib.cpp
[7/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\imgui_draw.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\imgui_draw.cpp
[8/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\imgui_widgets.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\imgui_widgets.cpp
[9/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_win32.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_win32.cpp
[10/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_glut.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_glut.cpp
[11/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_sdl.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_sdl.cpp
[12/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\imgui.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\imgui.cpp
[13/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_glfw.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_glfw.cpp
[14/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_allegro5.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_allegro5.cpp
[15/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_dx9.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_dx9.cpp
[16/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_dx10.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_dx10.cpp
[17/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_dx12.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_dx12.cpp
[18/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_metal.mm.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.mm
FAILED: CMakeFiles/imgui.dir/examples/imgui_impl_metal.mm.obj 
C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_metal.mm.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.mm
imgui_impl_metal.mm
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(14): error C2018: unknown character '0x40'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(15): error C2018: unknown character '0x40'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(15): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(15): error C2146: syntax error: missing ';' before identifier 'MTLDevice'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(17): error C2065: 'id': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(17): error C2065: 'MTLDevice': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(17): error C2065: 'device': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(21): error C2061: syntax error: identifier 'id'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(25): error C2065: 'id': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(25): error C2065: 'MTLDevice': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(25): error C2065: 'device': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(27): error C2065: 'id': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(27): error C2065: 'MTLDevice': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.h(27): error C2065: 'device': undeclared identifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_metal.mm(23): fatal error C1083: Cannot open type library file: 'Metal/Metal.h': No such file or directory
[19/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_osx.mm.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.mm
FAILED: CMakeFiles/imgui.dir/examples/imgui_impl_osx.mm.obj 
C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_osx.mm.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.mm
imgui_impl_osx.mm
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(13): error C2018: unknown character '0x40'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(14): error C2018: unknown character '0x40'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(18): error C2146: syntax error: missing ')' before identifier 'view'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(18): error C3646: 'view': unknown override specifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(18): error C2059: syntax error: ')'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(19): error C2146: syntax error: missing ')' before identifier 'event'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(19): error C3646: 'event': unknown override specifier
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(19): error C2143: syntax error: missing ';' before '*'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(19): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(19): error C2146: syntax error: missing ';' before identifier 'view'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.h(19): error C2059: syntax error: ')'
D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_osx.mm(13): fatal error C1083: Cannot open type library file: 'Cocoa/Cocoa.h': No such file or directory
[20/22] C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1427~1.291\bin\Hostx64\x64\cl.exe   /TP  -ID:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean -ID:\vcpkg\installed\x64-windows\include -ID:\vcpkg\installed\x64-windows\include\glad -ID:\vcpkg\installed\x64-windows\include\SDL2 -IC:\VulkanSDK\1.2.148.1\Include /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /FoCMakeFiles\imgui.dir\examples\imgui_impl_dx11.cpp.obj /FdCMakeFiles\imgui.dir\imgui.pdb /FS -c D:\vcpkg\buildtrees\imgui\src\v1.77-a2cb3da790.clean\examples\imgui_impl_dx11.cpp
ninja: build stopped: subcommand failed.
```

--- a/ports/imgui/CONTROL
+++ b/ports/imgui/CONTROL
@@ -1,6 +1,6 @@
 Source: imgui
 Version: 1.77
-Port-Version: 2
+Port-Version: 3
 Homepage: https://github.com/ocornut/imgui
 Description: Bloat-free Immediate Mode Graphical User interface for C++ with minimal dependencies.
 

--- a/ports/imgui/portfile.cmake
+++ b/ports/imgui/portfile.cmake
@@ -10,6 +10,10 @@ vcpkg_from_github(
 
 file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
 
+if(("metal-binding" IN_LIST FEATURES OR "osx-binding" IN_LIST FEATURES) AND (NOT VCPKG_TARGET_IS_OSX))
+    message(FATAL_ERROR "Feature metal-binding and osx-binding are only supported on osx.")
+endif()
+
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     allegro5-binding            IMGUI_BUILD_ALLEGRO5_BINDING
     dx9-binding                 IMGUI_BUILD_DX9_BINDING

